packages/driver/src/cross-origin/communicator.ts

```diff
@@ -107,6 +107,33 @@ export class PrimaryOriginCommunicator extends EventEmitter {
       data: preprocessedData,
     }, '*')
   }
+
+  /**
+   * Promisified event sent to the the primary communicator that expects the same event reflected back with the response.
+   * @param {string} event  - the name of the event to be sent.
+   * @param {Cypress.ObjectLike} data  - any meta data to be sent with the event.
+   * @param options - contains boolean to sync globals
+   * @returns the response from primary of the event with the same name.
+   */
+  toSpecBridgePromise<T> (originPolicy: string, event: string, data?: any) {
+    return new Promise<T>((resolve, reject) => {
+      let timeoutId
+
+      const handler = (result) => {
+        clearTimeout(timeoutId)
+        resolve(result)
+      }
+
+      timeoutId = setTimeout(() => {
+        this.off(event, handler)
+        reject()
+      }, 1000)
```

Comment:
I can reject this with a more descriptive error, but it will be up to the caller to appropriately translate and throw a useful error (or not) in the event that you can't communicate.

Reply:
with the error now I am wondering if we just want to have a variable like `const promisifiedTimeout = 1000` and just use a formatted string for the error so if we change it in the future we just do it it one place.

Reply:
I think an error message would be a good idea, even if the caller ends up ignoring it. Just in case this error leaks out somehow, it will be much easier to pinpoint the issue with an error message.

Reply:
Rejected with an error now!